fortnite_api/shop.py

```diff
@@ -1,266 +1,272 @@
-            'newDisplayAsset'] else None
-        self.items: List[Union[BrCosmetic, CarCosmetic, InstrumentCosmetic, LegoKitCosmetic, JamTrack]] = []
-        if data['brItems']:
-            self.items.extend([BrCosmetic(item) for item in data['brItems']])
-        if data['cars']:
-            self.items.extend([CarCosmetic(item) for item in data['cars']])
-        if data['instruments']:
-            self.items.extend([InstrumentCosmetic(item) for item in data['instruments']])
-        if data['legoKits']:
-            self.items.extend([LegoKitCosmetic(item) for item in data['legoKits']])
-        if data['tracks']:
-            self.items.extend([JamTrack(item) for item in data['tracks']])
-        self.raw_data: dict = data
-
-
-class ShopBundle:
-
-    def __init__(self, data: dict):
+from .abc import Hashable
+from .asset import Asset
+from .cosmetics import CosmeticBr, CosmeticCar, CosmeticInstrument, CosmeticLegoKit, CosmeticTrack
+from .http import HTTPClientT
+from .material import MaterialInstance
+from .utils import get_with_fallback, parse_time
+
+
+class ShopEntryBundle(Generic[HTTPClientT]):
+    """Represents a shop entry bundle.
+
+    Attributes
+    ----------
+    name: :class:`str`
+        The name of the bundle.
+    info: :class:`str`
+        The information of the bundle.
+    image: :class:`Asset`
+        The image of the bundle.
+    """
+
+    __slots__: Tuple[str, ...] = ('name', 'info', 'image')
+
+    def __init__(self, *, data: Dict[str, Any], http: HTTPClientT) -> None:
         self.name: str = data['name']
         self.info: str = data['info']
-        self.image_url: str = data['image']
-        self.raw_data: dict = data
+        self.image: Asset[HTTPClientT] = Asset(url=data['image'], http=http)
 
 
-class ShopBanner:
+class ShopEntryBanner:
+    """A class to represent a shop entry banner.
 
-    def __init__(self, data: dict):
+    Attributes
+    ----------
+    value: :class:`str`
+        The content of the banner.
+    intensity: :class:`str`
+        The visual style or intensity of the banner.
+    backend_value: :class:`str`
+        The backend value of the banner.
+    """
+
+    __slots__: Tuple[str, ...] = ('value', 'intensity', 'backend_value')
+
+    def __init__(self, *, data: Dict[str, Any]) -> None:
         self.value: str = data['value']
         self.intensity: str = data['intensity']
         self.backend_value: str = data['backendValue']
-        self.raw_data: dict = data
 
 
-class ShopLayout:
+class ShopEntryLayout(Hashable, Generic[HTTPClientT]):
+    """Represents the layout of a shop entry.
+
+    Attributes
+    ----------
+    id: :class:`str`
+        The ID of the layout.
+    name: :class:`str`
+        The name of the layout.
+    category: Optional[:class:`str`]
+        The category of the layout, if any.
+    index: :class:`int`
+        The index of the layout.
+    show_ineligible_offers: :class:`str`
+        The show ineligible offers flag of the layout.
```

Comment:
I think this need to be better worded. Something like `Whether ineligible offers are displayed in the layout`